src/directive/rateLimit.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  defaultKeyGenerator,
+  rateLimitDirective,
+} from 'graphql-rate-limit-directive';
+import { RateLimiterMemory } from 'rate-limiter-flexible';
+import { GraphQLError } from 'graphql';
+
+export let rateLimiter: CustomRateLimiterMemory;
+
+export class CustomRateLimiterMemory extends RateLimiterMemory {
+  constructor(props) {
+    super(props);
+
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    rateLimiter = this;
+  }
+
+  // for debugging purposes
+  consume(key, pointsToConsume, options) {
```

Comment:
Can we delete it now I guess?
Or at least comment

Reply:
I wanted to keep this so future devs would have an easy time debugging in case they stumble upon this case. Though I'm open to removing this as a whole.

Reply:
It's still active?

Reply:
Have made a better comment now for easier understanding of its purpose.